### PR TITLE
Use benchmark.js

### PR DESCRIPTION
**Pull request by ryanio**
This PR targets merging into #794 to:
* Use `vm.runBlock()` instead of looping over txs with `vm.runTx()`
* Use benchmark.js to profile runs
* Use [github-action-benchmark](https://github.com/rhysd/github-action-benchmark) to store and visualize data in GH Pages.
* Reduce fixture file from 50 to 15 blocks (80mb -> 30mb)
  * ci set for 10 blocks i think is sufficient for now

Sample output:

```
Total number of blocks in data set: 15
Number of blocks to sample: 10
Block 9422905 x 5,458 ops/sec ±3.03% (83 runs sampled)
Block 9422906 x 5,603 ops/sec ±3.14% (90 runs sampled)
Block 9422907 x 5,149 ops/sec ±5.38% (81 runs sampled)
Block 9422908 x 5,526 ops/sec ±0.92% (90 runs sampled)
Block 9422909 x 5,394 ops/sec ±0.72% (87 runs sampled)
Block 9422910 x 4,619 ops/sec ±6.22% (81 runs sampled)
Block 9422911 x 4,944 ops/sec ±2.30% (84 runs sampled)
Block 9422912 x 4,713 ops/sec ±1.66% (83 runs sampled)
Block 9422913 x 4,763 ops/sec ±1.28% (86 runs sampled)
Block 9422914 x 4,536 ops/sec ±1.93% (82 runs sampled)
Block 9422915 x 4,346 ops/sec ±1.98% (79 runs sampled)
```

Pages link: http://ethereumjs.github.io/ethereumjs-vm/dev/bench/

### CODE REVIEW

**Comment by holgerd77** [comment]
Super cool, the tool suite you figured out with this `benchmark` action and `benchmark.js` looks really nice, seems to fit exactly our needs. 😄

File: .github/workflows/vm-build.yml [hunk 1]
```diff
@@ -105,3 +105,47 @@ jobs:
...
+        uses: actions/cache@v2
+        id: cache
+        with:
+          key: VM-${{ runner.os }}-${{ matrix.node-version }}-${{ hashFiles('**/package-lock.json') }}
+          path: '**/node_modules'
+
+      # Installs root dependencies, ignoring Bootstrap All script.
+      # Bootstraps the current package only
+      - run: npm install --ignore-scripts && npm run bootstrap:vm
+        if: steps.cache.outputs.cache-hit != 'true'
```
Comment (holgerd77) [inline .github/workflows/vm-build.yml:127]: Just out of interest: what is this line doing respectively how does this behave?

File: packages/vm/scripts/benchmarks/mainnetBlocks.ts [hunk 2]
```diff
@@ -1,76 +1,56 @@
...
 import { StateManager, DefaultStateManager } from '../../dist/state'
 import BN = require('bn.js')
+import Benchmark = require('benchmark')
+
+const suite = new Benchmark.Suite()
 
 async function main() {
   const args = process.argv
-  if (args.length < 3 || args.length > 4) {
+  if (args.length !== 3 && args.length !== 4) {
```
Comment (holgerd77) [inline packages/vm/scripts/benchmarks/mainnetBlocks.ts:15]: Not a big deal but just wonder why you changed that evaluation? Code before was more generic and more flexibly allowed for range checks? 🤔

But again, in the end I really don't care, was just wondering. 😄

Comment (ryanio) [reply packages/vm/scripts/benchmarks/mainnetBlocks.ts:15]: oh i was just tinkering around with the script and found that logic easier to parse ;p i can change it back if you want!

File: .github/workflows/vm-build.yml (hunk 1 above)
Comment (ryanio) [reply .github/workflows/vm-build.yml:127]: installs deps and builds the vm for use in the benchmarks in the next steps

Comment (holgerd77) [reply .github/workflows/vm-build.yml:127]: Ah no, was just referring to the `if`-clause: `if: steps.cache.outputs.cache-hit != 'true'` 😄

File: packages/vm/scripts/benchmarks/mainnetBlocks.ts (hunk 2 above)
Comment (holgerd77) [reply packages/vm/scripts/benchmarks/mainnetBlocks.ts:15]: Don't worry, will mark as resolved. :-)

File: .github/workflows/vm-build.yml (hunk 1 above)
Comment (ryanio) [reply .github/workflows/vm-build.yml:127]: oh :) that’ll skip the install step if it was able to restore the deps from the cache step above it

**Comment by ryanio** [comment]
This is now ready for review to merge into #794 and then would like to merge into master.

**Review by holgerd77 — APPROVED** [review]
Looks good

**Pull request merged**